pkg/apis/apps/types.go

```diff
@@ -187,6 +187,12 @@ type StatefulSetStatus struct {
 	// updateRevision, if not empty, indicates the version of the StatefulSet used to generate Pods in the sequence
 	// [replicas-updatedReplicas,replicas)
 	UpdateRevision string
+
+	// collisionCount is the count of hash collisions for the StatefulSet. The StatefulSet controller
+	// uses this field as a collision avoidance mechanism when it needs to create the name for the
+	// newest ControllerRevision.
+	// +optional
+	CollisionCount *int64
```

Comment:
Why is this int64 and not int32?  Do you expect more than 2 billion collisions?  JSON is known to have trouble carrying int64 values, which can make automated API fuzzing difficult.